daemon_lib.py

```diff
@@ -17,60 +17,70 @@
 
 
 def get_address() -> str:
-    """Get address of the daemon."""
+    """Get address of the BlenderKit-Client."""
     return f"http://127.0.0.1:{get_port()}"
 
 
 def get_port() -> str:
-    """Get the most probable port of currently running daemon.
+    """Get the most probable port of currently running BlenderKit-Client.
     After add-on registration and if all goes well, the port is the same as
     """
-    return global_vars.DAEMON_PORTS[0]
+    return global_vars.CLIENT_PORTS[0]
 
 
-def reorder_ports(port: str):
-    """Reorder DAEMON_PORTS so the specified port is first."""
-    i = global_vars.DAEMON_PORTS.index(port)
-    global_vars.DAEMON_PORTS = (
-        global_vars.DAEMON_PORTS[i:] + global_vars.DAEMON_PORTS[:i]
-    )
-
-
-def get_daemon_directory_path() -> str:
-    """Get path to daemon directory in blenderkit_data directory."""
-    global_dir = bpy.context.preferences.addons["blenderkit"].preferences.global_dir
-    directory = path.join(global_dir, "daemon")
-    return path.abspath(directory)
+def ensure_minimal_data(data: dict = {}) -> dict:
+    """Ensure that the data send to the BlenderKit-Client contains:
+    - app_id is the process ID of the Blender instance, so BlenderKit-client can return reports to the correct instance.
+    - api_key is the authentication token for the BlenderKit server, so BlenderKit-Client can authenticate the user.
+    - addon_version is the version of the BlenderKit add-on, so BlenderKit-client has understanding of the version of the add-on making the request.
+    """
+    if "app_id" not in data:
+        data["app_id"] = os.getpid()
+    if "api_key" not in data:
+        data["api_key"] = bpy.context.preferences.addons[
+            "blenderkit"
+        ].preferences.api_key
+    if "addon_version" not in data:
+        data["addon_version"] = (
+            f"{global_vars.VERSION[0]}.{global_vars.VERSION[1]}.{global_vars.VERSION[2]}.{global_vars.VERSION[3]}"
+        )
+    if "platform_version" not in data:
+        data["platform_version"] = platform.platform()
```

Comment:
Just in case you didn't know, but feel free to ignore...

```suggestion
    data.setdefault("app_id", os.getpid())
    data.setdefault("api_key", bpy.context.preferences.addons[
        "blenderkit"
    ].preferences.api_key)
    data.setdefault("addon_version",
        f"{global_vars.VERSION[0]}.{global_vars.VERSION[1]}.{global_vars.VERSION[2]}.{global_vars.VERSION[3]}"
    )
    data.setdefault("platform_version", platform.platform())
```

Reply:
🥇